apps/web/src/workers/readMetadata.ts

```diff
@@ -0,0 +1,72 @@
+import { readMetadata } from '@latitude-data/compiler'
+import { type DocumentVersion } from '@latitude-data/core/browser'
+
+export type ReadMetadataWorkerProps = Parameters<typeof readMetadata>[0] & {
+  document?: DocumentVersion
+  documents?: DocumentVersion[]
+}
+
+self.onmessage = async function (event: { data: ReadMetadataWorkerProps }) {
+  const { document, documents, prompt, ...rest } = event.data
+  const referenceFn = readDocument(document, documents, prompt)
+  const metadata = await readMetadata({
+    ...rest,
+    prompt,
+    referenceFn: referenceFn ?? undefined,
+  })
```

Comment:
a tradeoff is that we don't have configSchema validation since it depends on zod which i can't manage to get it working on web workers